module-services/service-cellular/src/ServiceCellularPriv.cpp

```diff
@@ -390,4 +394,98 @@ namespace cellular::internal
             return false;
         };
     }
+
+    void ServiceCellularPriv::initCSQHandler()
+    {
+        csqHandler->onEnableCsqReporting = [this]() {
+            auto handle = owner->getTaskHandle();
+            if (owner->cpuSentinel) {
+                owner->cpuSentinel->HoldMinimumFrequencyAndWait(bsp::CpuFrequencyMHz::Level_4, handle, 2000);
```

Comment:
Maybe we could move the magic value to constexpr

Reply:
done